src/handlers/web3.ts

```diff
@@ -698,53 +703,51 @@ export const getDataForNftTransfer = async (
   from: string,
   to: string,
   asset: ParsedAddressAsset
-): Promise<string> => {
-  const nftVersion = asset.asset_contract?.nft_version;
-  const schema_name = asset.asset_contract?.schema_name;
-  if (nftVersion === '3.0') {
-    const transferMethodHash = smartContractMethods.nft_transfer_from.hash;
-    const data = ethereumUtils.getDataString(transferMethodHash, [
-      ethereumUtils.removeHexPrefix(from),
+): Promise<string | undefined> => {
+  if (!asset.id || !asset.asset_contract?.address) return;
+  const lowercasedContractAddress = asset.asset_contract.address.toLowerCase();
+  const standard = asset.asset_contract?.schema_name;
+  let data;
+  if (
+    lowercasedContractAddress === CRYPTO_KITTIES_NFT_ADDRESS &&
+    asset.network === Network.mainnet
+  ) {
+    const transferMethod = smartContractMethods.token_transfer;
+    data = ethereumUtils.getDataString(transferMethod.hash, [
       ethereumUtils.removeHexPrefix(to),
       convertStringToHex(asset.id),
     ]);
-    return data;
-  } else if (schema_name === 'ERC1155') {
-    const transferMethodHash =
-      smartContractMethods.erc1155_safe_transfer_from.hash;
-    const data = ethereumUtils.getDataString(transferMethodHash, [
+  } else if (
+    lowercasedContractAddress === CRYPTO_PUNKS_NFT_ADDRESS &&
+    asset.network === Network.mainnet
+  ) {
+    const transferMethod = smartContractMethods.punk_transfer;
+    data = ethereumUtils.getDataString(transferMethod.hash, [
+      ethereumUtils.removeHexPrefix(to),
+      convertStringToHex(asset.id),
+    ]);
+  } else if (standard === ERC1155) {
+    const transferMethodHash = smartContractMethods.erc1155_transfer.hash;
+    data = ethereumUtils.getDataString(transferMethodHash, [
       ethereumUtils.removeHexPrefix(from),
       ethereumUtils.removeHexPrefix(to),
       convertStringToHex(asset.id),
       convertStringToHex('1'),
       convertStringToHex('160'),
       convertStringToHex('0'),
     ]);
-    return data;
-  } else if (IS_TESTING === 'true') {
-    const transferMethodHash = smartContractMethods.nft_transfer.hash;
-    const data = ethereumUtils.getDataString(transferMethodHash, [
+  } else {
+    // ERC721s + unsupported NFTs (will most likely fail)
+    if (standard !== ERC721) {
+      logger.warn(`Attempting to send unsupported NFT:\n${asset}`);
+    }
+    const transferMethod = smartContractMethods.erc721_transfer;
+    data = ethereumUtils.getDataString(transferMethod.hash, [
+      ethereumUtils.removeHexPrefix(from),
       ethereumUtils.removeHexPrefix(to),
       convertStringToHex(asset.id),
     ]);
-    return data;
   }
-
-  const address = asset.asset_contract?.address!;
-  const abi = await fetchContractABI(address);
-  const iface = new Interface(abi);
-
-  const isTransferFrom =
```

Comment:
this is what is causing us problems

what's happening here:
1. get contract abi from address
  * the abi defines how you can interact with the contract. in this case, that means transfer the nft.
2. check if abi defines `nft_transfer_from`
3. create tx data using`nft_transfer_from` if available, else `nft_transfer`

problems w/ this code:
* proxy contracts don't directly expose the abi - even if an nft supports `nft_transfer_from`, the abi might not say so and this check will be a false negative
* `nft_transfer` simply is not a standard nft transfer method, it's primarly used for ERC20s

why some nft sends fail:
* if an nft is an erc721 w/ a proxy contract, we incorrectly determine that it doesn't support `nft_transfer_from` and instead try to use `nft_transfer` which will basically *never* work